packages/frontend/core/src/modules/journal/store/journal.ts

```diff
@@ -11,6 +11,17 @@ export class JournalStore extends Store {
     super();
   }
 
+  allJournalDates$ = LiveData.computed(get => {
+    return new Set(
+      get(this.docsService.list.docs$)
+        .filter(doc => {
+          const journal = get(doc.properties$.selector(p => p.journal));
+          return !!journal && isJournalString(journal);
+        })
+        .map(doc => doc.properties$.value.journal)
```

Comment:
get(doc.properties$).journal?

Reply:
Not really sure, I think that should already be observed inside `filter`. It seems to heavy here.

Reply:
I think a well designed `computed` should be able to batch the updates